docs/user_manual.md

```diff
@@ -49,40 +60,277 @@ python -m virtualenv venv
+  ``` bash
+  pip uninstall pycocotools
+  pip install pycocotools --no-binary=pycocotools
+  ```
+
+- When installing directly from the repository, you can change the
+  installation branch with `...@<branch_name>`. Also use `--force-reinstall`
+  parameter in this case. It can be useful for testing of unreleased
+  versions from GitHub pull requests.
+
+## How to use Datumaro
+
+As a standalone tool or a Python module:
 
 ``` bash
 datum --help
-```
-
-As a python module:
-> The directory containing Datumaro should be in the `PYTHONPATH`
-> environment variable or `cvat/datumaro/` should be the current directory.
 
-``` bash
 python -m datumaro --help
 python datumaro/ --help
 python datum.py --help
 ```
 
-As a python library:
+As a Python library:
 
 ``` python
-import datumaro
+from datumaro.components.project import Project
+from datumaro.components.dataset import Dataset
+from datumaro.components.extractor import Label, Bbox, DatasetItem
+...
+dataset = Dataset.import_from(path, format)
+...
+```
+
+### Glossary
+
+- Basic concepts:
+  - dataset - A number of media (dataset items) and associated annotations
+  - project - A combination of multiple datasets, plugins, models and metadata
+
+- Project versioning concepts:
+  - data source - a link to a dataset or a copy of a dataset inside a project.
+    Basically, an URL + dataset format name
+  - project revision - a commit hash or a named reference from
+    Git (branch, tag, HEAD~3 etc.).
+  - working / head / revision tree - a project build tree and plugins at
+    a specified revision
+  - data source revision - a data source hash at a specific stage
+  - object - a tree or a data source revision data
+
+- Dataset path concepts: <a id="revpath"></a>
+  - source / dataset / revision / project path - a path to a dataset in a
+    special format
+
+    - (project local) **rev**ision **path**s - a way to specify the path
+      to a source revision in the CLI, the syntax is:
+      `<revision>:<source/target name>`, any part can be omitted.
+      - Default revision is the working tree of the project
+      - Default target is the compiled project
+
+    - dataset revpath - a path to a dataset in the following format:
+      `<dataset path>:<format>`
+      - Format is optional. If not specified, will try to autodetect
+
+    - full revpath - a path to a source revision in a project, the syntax is:
+      `<project path>@<revision>:<target name>`, any part can be omitted.
+      - Default project is the current project (`-p`/`--project` CLI arg.)
+      - Default revision is the working tree of the project
+      - Default target is the compiled project
+
+- Dataset building concepts:
+  - stage - a modification of a data source. A transformation,
+    filter or something else.
+  - build tree - a directed graph (tree) with leaf nodes at data sources
+    and a single root node called "project"
+  - build target - a data source or a stage
+  - pipeline - a subgraph of a build target
```

Comment:
This definition doesn't make sense. A build target is a single node. It doesn't have subgraphs.